nixos/modules/services/networking/kresd.nix

```diff
@@ -137,6 +188,18 @@ in {
       CacheDirectoryMode = "0770";
     };
     # We don't mind running stop phase from wrong version.  It seems less racy.
-    systemd.services."kresd@".stopIfChanged = false;
+    systemd.services.knot-resolver.stopIfChanged = false;
+
+
+    # TODO: also install shell completions, maybe man pages, etc.
+    environment.systemPackages = [
+      (pkgs.runCommandLocal "knot-resolver-cmds"
+        { buildInputs = [ pkgs.makeWrapper ]; }
+        # Wrapping, as config might've changed the location of the management socket.
+        ''
+          makeWrapper '${manager}/bin/kresctl' "$out/bin/kresctl" \
+            --add-flags --config=${json}
```

Comment:
Why not take the config from /etc/ instead?

Reply:
I don't know.  The etc thing is just a symlink to this file.  I thought it better to reference the target directly, but no hard feelings about that.  There are more places like this and a "not required" comment on the etc file (just there for users to inspect right now).